params/json.libevm.go

```diff
@@ -39,36 +39,54 @@ type chainConfigWithExportedExtra struct {
 	Extra                      *pseudo.Type `json:"extra"` // `c.extra` is otherwise unexported
 }
 
-// UnmarshalJSON implements the [json.Unmarshaler] interface.
-func (c *ChainConfig) UnmarshalJSON(data []byte) error {
-	switch reg := registeredExtras; {
-	case reg.Registered() && !reg.Get().reuseJSONRoot:
-		return c.unmarshalJSONWithExtra(data)
-
-	case reg.Registered() && reg.Get().reuseJSONRoot: // although the latter is redundant, it's clearer
-		c.extra = reg.Get().newChainConfig()
-		if err := json.Unmarshal(data, c.extra); err != nil {
-			c.extra = nil
-			return err
-		}
-		fallthrough // Important! We've only unmarshalled the extra field.
-	default: // reg == nil
+// UnmarshalJSON implements the [json.Unmarshaler] interface and JSON decodes
+// `data` according to the following:
+//   - extra is not registered:
+//     `data` is decoded into `c` and the extra is ignored.
+//   - extra is registered and the registered reuseJSONRoot field is false:
+//     `data` is decoded into `c` and the "extra" JSON field in `data` is decoded into `c.extra`.
+//   - extra is registered and the registered reuseJSONRoot field is true:
+//     `data` is decoded into `c` and `data` is decoded into `c.extra`.
+func (c *ChainConfig) UnmarshalJSON(data []byte) (err error) {
+	if !registeredExtras.Registered() {
+		// assume there is no extra
 		return json.Unmarshal(data, (*chainConfigWithoutMethods)(c))
 	}
+	extraConstructors := registeredExtras.Get()
```

Comment:
(style) Please don't use such verbose variable names when the scope is only 3 lines; they distract from reading of the code. I'd personally be happy with just `r`, but understand that others are not—as long as it's not 5 syllables.

[See style guide](https://google.github.io/styleguide/go/decisions#:~:text=The%20general%20rule,avoid%20extraneous%20information.)

Reply:
I've read a few style guides including that one, my view on it, unless it's a really obvious use (i.e. `i` for a range index), it's nice to have a meaningful name almost always. For example `c` instead of `extraConstructors` made me wonder what was `c` and had to lookup where it was created and what `.Get()` was returning, and, because it's a generic function, had to dig into `extraConstructors struct`. Not a big deal obviously, but it does make readability worst in my opinion.

Obviously that's my opinion on this, so I'll conform to the existing code and use shorter variable names 😉